### PR TITLE
[Dev 213518] - ios example integrates with backend to generate requestsignature

### PR DESCRIPTION
**Pull request by marcosrivereto**

### CODE REVIEW

**Review by andresguedes — APPROVED** [review]
👍🏽

File: TrustlySDKDemoIOS/TrustlyLightboxViewController.swift [hunk 1]
```diff
@@ -16,12 +16,43 @@ protocol TrustlyLightboxViewProtocol {
 
 class TrustlyLightboxViewController: UIViewController {
     
+    let signatureApi = SignatureAPI()
```
Comment (lukevance) [inline TrustlySDKDemoIOS/TrustlyLightboxViewController.swift:19]: maybe consider "requestSignatureAPI" just to keep the nomenclature consistent?

File: TrustlySDKDemoIOS/TrustlyLightboxViewController.swift [hunk 2]
```diff
@@ -16,12 +16,43 @@ protocol TrustlyLightboxViewProtocol {
...
     var delegate: TrustlyLightboxViewProtocol?
     
+    var spinner = UIActivityIndicatorView(style: .whiteLarge)
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        self.showSpinner()
+        
+        self.updateEstablishWithRequestSignature()
```
Comment (lukevance) [inline TrustlySDKDemoIOS/TrustlyLightboxViewController.swift:30]: 👍

File: TrustlySDKDemoIOS/TrustlyLightboxViewController.swift [hunk 3]
```diff
@@ -34,6 +65,26 @@ class TrustlyLightboxViewController: UIViewController {
...
 }
+
+//MARK: Network extension
+extension TrustlyLightboxViewController {
+    
+    func updateEstablishWithRequestSignature() {
+        
+        signatureApi.generateRequestSignatureFor(establishData: self.establishData) { (result) in
+            do {
+                try self.establishData["requestSignature"] = result.get()
```
Comment (lukevance) [inline TrustlySDKDemoIOS/TrustlyLightboxViewController.swift:80]: I'm not sure that this is the right place, but can you comment out the implementation of signing the request and leave a comment to point to where a user can enable it? It's important that a user can follow two paths:

1. simple quickstart: just run the project with no backend integration and no requestSignature
2. add security: later when they have solved this part, the user may want to revisit the project to understand how to setup requestSignatures

See the [AC here ](https://trustly.atlassian.net/browse/DEV-213518) that mentions "uncommenting" the logic. That may not be the right approach but the user should easily understand how to go from step 1 above to step 2.

File: TrustlySDKDemoIOS/Utils/Constants.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//
+//  Constants.swift
+//  TrustlySDKDemoIOS
+//
+//  Created by Marcos Rivereto on 30/11/23.
+//
+
+struct Constants {
+    static let BASE_URL = "https://shop-on-merchant.fly.dev"
```
Comment (lukevance) [inline TrustlySDKDemoIOS/Utils/Constants.swift:9]: Should this instead be "localhost" ? Also maybe leave a comment to explain the connection to the nestjs example app and also please add a note to the ReadMe similar to this one: https://github.com/TrustlyInc/trustly-webview-example/blob/4adbfa6713848502616cac99d493b9ded293972b/react-web-app/README.md#server-side-features

**Review by lukevance — CHANGES_REQUESTED** [review]
Looks awesome and ran smoothly! But please see the notes regarding instructions for the user and how this should be a optional feature which the user can enable instead of a requirement.

File: TrustlySDKDemoIOS/TrustlyLightboxViewController.swift (hunk 1 above)
Comment (marcosrivereto) [reply TrustlySDKDemoIOS/TrustlyLightboxViewController.swift:19]: 👍

File: TrustlySDKDemoIOS/Utils/Constants.swift (hunk 4 above)
Comment (marcosrivereto) [reply TrustlySDKDemoIOS/Utils/Constants.swift]: Updated with the **_Server Side Feature_** section, and text about what url we should use.

File: TrustlySDKDemoIOS/TrustlyLightboxViewController.swift (hunk 3 above)
Comment (marcosrivereto) [reply TrustlySDKDemoIOS/TrustlyLightboxViewController.swift]: I already commented the code and added some instructions about when the merchant need to use the request signature function, and update the ReadMe file with some instructions too.

**Review by andresguedes — APPROVED** [review]

**Review by pedroabdenor — APPROVED** [review]
👍

File: README.md [hunk 5]
```diff
@@ -6,42 +6,12 @@ This example app provides a simple integration of the Trustly Lightbox iOS SDK i
 
 - clone
 - open project
-- Install Trustly SDK - This can be done 3 ways
+- Install Trustly SDK - This can be done 2 ways
     - Cocoapods
     - Swift Package Manager
-    - Manually Download
```
Comment (lukevance) [inline README.md:12]: Is this because we no longer want to generate packages?

File: README.md [hunk 6]
```diff
@@ -74,10 +44,48 @@ For production choose the `main` branch, but in order to develop or test against
...
 
-*****
-
 In the `MerchantViewController` file, you will find a controller simulating the a merchant app screen. This controller will call the `TrustlyLightboxViewController` class, and this class will be responsible to call the `PayWithMyBank SDK`.
 
 Note that the application has a URL Scheme configured in order for the app to have a Deep Link of `demoapp://`. A Deep Link is required for a successful Trustly Lightbox integration within a mobile app.
 
+# Define Establish Data with a Request Signature
+
+As the Trustly Lightbox SDK runs on the client, requests between it and the Trustly API must be secured. Calculate a requestSignature using your Access Key from your server and fetch it from your iOS app prior to rendering the Trustly Lightbox. For more information read the section `Define Establish Data with a Request Signature` in this [documentation](https://amer.developers.trustly.com/payments/docs/ios-quickstart#define-establish-data-with-a-request-signature), and check the code snipets bellow in the `TrustlyLightboxViewController.swift` file.
```
Comment (lukevance) [inline README.md:53]: Nice 👍

File: README.md [hunk 7]
```diff
@@ -74,10 +44,48 @@ For production choose the `main` branch, but in order to develop or test against
...
+
+```swift
+        ...
+
+        /* Uncomment this function only if your merchant setup has the "Extended Security" enable in Admin console, and uncomment the code between the lines
```
Comment (lukevance) [inline README.md:58]: 👍

File: TrustlySDKDemoIOS/Utils/Constants.swift [hunk 8]
```diff
@@ -0,0 +1,11 @@
+//
+//  Constants.swift
+//  TrustlySDKDemoIOS
+//
+//  Created by Marcos Rivereto on 30/11/23.
+//
+
+struct Constants {
+    static let BASE_URL = "<YOUR BACKEND URL>"
```
Comment (lukevance) [inline TrustlySDKDemoIOS/Utils/Constants.swift:9]: 👍

**Review by lukevance — APPROVED** [review]
LGTM!

**Pull request merged**